src/dag-node.js

```diff
@@ -51,7 +51,7 @@ module.exports = class DAGNode {
     // ensure links are instances of DAGLink
     if (links) {
       links.forEach((l) => {
-        if (l.name && typeof l.toJSON === 'function') {
+        if (l instanceof DAGLink) {
```

Comment:
Yeah, I just ran into this the other day. Let's check `constructor` instead -- I think that will be more reliable.